### PR TITLE
added Boyer-Moore-algorithm for Strings

### PR DESCRIPTION
**Pull request by Hamza1821**
## 📥 Pull Request
added Boyer-Moore-algorithm for Strings
### 
### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Hamza1821** [comment]
sir please merge this

**Comment by Hamza1821** [comment]
please merge it sir

**Comment by ajay-dhangar** [comment]
> [Hamza1821](/Hamza1821)

Can you share with me the issue number or a link to the issue that you are referring to?

**Comment by Hamza1821** [comment]
Sir i didnot created the issue for it

File: package.json [hunk 1]
```diff
@@ -23,6 +23,7 @@
     "@giscus/react": "^3.0.0",
     "@heroicons/react": "^2.1.5",
     "@mdx-js/react": "^3.0.0",
+    "algo": "file:",
```
Comment (ajay-dhangar) [inline package.json]: Why did you add line 26? I think it should be deleted.

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
`package.json` and `package-lock.json` are automatically generated, so don't edit them.

File: package.json (hunk 1 above)
Comment (Hamza1821) [reply package.json]: I have not added it

Comment (ajay-dhangar) [reply package.json]: you need delete line number 26 and repush

**Comment by Hamza1821** [comment]
Done now please merge

**Comment by ajay-dhangar** [comment]
I see also, lock file

{265D6052-D8D9-4197-90B3-C89F2EA350E9}

Can you please check out your changes and fix this? The file is too large for me to see the changes directly.

**Pull request closed without merging**